docs/advanced-usage/naming-generated-files.md

```diff
@@ -0,0 +1,53 @@
+---
+title: Naming generated files
+weight: 11
+---
+
+### Naming conversion files
+
+By default, all conversion files will be named in this format:
+
+```
+{original-file-name-without-extension}-{name-of-the-conversion}.{extension}
+```
+
+Should you want to name your conversion file using another format,
+then you can specify the class name of your own `FileNamer` in the `file_namer` key
+of the `media-library.php` config file.
+
+The only requirements is that your class extends `Spatie\MediaLibrary\Support\FileNamer`.
+In your class you should implement 2 methods.
+1. `getFileName` that returns the name of the file without the extension. You can prefix or suffix this part.
+2. `getConversionFileName` that returns the file name combined with the conversion name, by default.
+
+Here the implementation of `Spatie\MediaLibrary\Conversion\DefaultConversionFileNamer`
+
+```php
+namespace Spatie\MediaLibrary\Support\FileNamer;
+
+use Spatie\MediaLibrary\Conversions\Conversion;
+
+class DefaultFileNamer extends FileNamer
+{
+    public function getFileName(string $fileName): string
+    {
+        return pathinfo($fileName, PATHINFO_FILENAME);
+    }
+
+    public function getConversionFileName(string $fileName, Conversion $conversion): string
+    {
+        return "{$fileName}-{$conversion->getName()}";
+    }
+}
+```
+
+### Naming responsive image files
+
+By default, all conversion files will be named in this format:
+
+```
+{original-file-name-without-extension}___{name-of-the-conversion}_{width}_{height}.{extension}
+```
+
+Just like the conversion file names, you can use another format for naming your files.
```

Comment:
This is unclear. So `getConversionFileName` is used as the basis for the responsive image file names?

Reply:
`DefaultFileNamer` needs 2 methods:
- `getFileName` used by conversions and responsive images. For example, a timestamp prefix can be added here. This will affect both the conversion file names and the responsive image file names.
- `getConversionFileName` is only used for conversions. For example `file-name---thumb` can be renamed to `file-name_thumb`.

Reply:
I still don't understand this fully.  To me it seems strange that `getFileName` is also used for conversion files, as `getConversionFileName` already handles that. It might be best to add a couple very explicit examples to the docs.

Reply:
I see what you mean. I'll have another look at a different solution

Reply:
Updated:

 `getFileName` is removed from the FileNamer. It needs these 2 methods (which do not interfere each other):
- `getConversionFileName`
- `getResponsiveImageFileName`